daemon/daemon.go

```diff
@@ -82,21 +79,22 @@ func (d *Daemon) Main(serve func(string, string)) int {
 // client.
 func (d *Daemon) Spawn() error {
 	binPath := d.BinPath
-	// Determine binPath.
-	if binPath == "" {
-		if len(os.Args) > 0 && path.IsAbs(os.Args[0]) {
-			binPath = os.Args[0]
-		} else {
-			// Find elvish in PATH
-			paths := strings.Split(os.Getenv("PATH"), ":")
-			result, err := util.Search(paths, "elvish")
-			if err != nil {
-				return errors.New("cannot find elvish: " + err.Error())
-			}
-			binPath = result
+	if !filepath.IsAbs(binPath) {
+		var err error
+		binPath, err = exec.LookPath(os.Args[0])
+		if err != nil {
+			return errors.New("cannot find elvish: " + err.Error())
 		}
 	}
-	return forkExec(nil, 0, binPath, d.DbPath, d.SockPath, d.LogPathPrefix)
+
+	cmd := exec.Command(binPath, "-daemon",
+		"-forked", "0",
+		"-bin", binPath,
+		"-db", d.DbPath,
+		"-sock", d.SockPath,
+		"-logprefix", d.LogPathPrefix,
+	)
+	return cmd.Run()
```

Comment:
I think it's better to understand what is happening under the scene than using a different interface to do the job :) According to the [doc](http://godoc.org/os/exec#Cmd.Run) `cmd.Run` waits for the subprocess to terminate, that is probably some step we should add.